### PR TITLE
Feat(filter): handleAbstractAbiDecode

### PR DESCRIPTION
**Pull request by Quazia**
The general idea here is to search all of the input string for the matching ABI function selector, and then apply the filter in that context. Still needs testing but this is the approximate approach I'm taking right now. It's possible I'll have to back-track during testing though.

### CODE REVIEW

File: src/filter/filters.ts [hunk 1]
```diff
@@ -166,6 +166,32 @@ export const handleAbiDecode = (context: any, filter: { $abi: Abi }) => {
   }
 }
 
+export const handleAbstractAbiDecode = (context: any, filter: any) => {
```
Comment (kmairin) [inline src/filter/filters.ts:169]: can `filter` and `context` be typed? I think we should avoid using `any` in a function signature

Comment (jimobrien) [reply src/filter/filters.ts:169]: at a minimum it looks like we can add `$abiAbstract` to filter

**Review by jimobrien — APPROVED** [review]

**Comment by Quazia** [comment]
@kmairin Ok - I refactored everything to be more tightly typed and am using a Map now - still need to do some testing to make sure I didn't break it in the process but how's this looking?

File: src/filter/filters.ts [hunk 2]
```diff
@@ -166,6 +178,39 @@ export const handleAbiDecode = (context: any, filter: { $abi: Abi }) => {
...
 }
 
+export const handleAbstractAbiDecode = (
+  context: any,
+  filter: { $abiAbstract?: Abi },
+) => {
+  const decodedReturn: ReturnType<typeof handleAbiDecode>[] = []
+  const elementCount = filter.$abiAbstract!.length
+  const contextMap = new Map<string, number>()
+  for (let i = 0; i < context.length - 6; i++) {
```
Comment (kmairin) [inline src/filter/filters.ts:188]: could we add a comment why it has to be `context.length - 6`?

File: src/filter/filters.ts [hunk 3]
```diff
@@ -166,6 +178,39 @@ export const handleAbiDecode = (context: any, filter: { $abi: Abi }) => {
...
+    contextMap.set(context.substring(i, i + 6), i)
+  }
+  for (let i = 0; i < elementCount; i++) {
+    const abiItem = filter.$abiAbstract![i]
+    if (abiItem.type === 'function') {
+      const functionSelector = getFunctionSelector(abiItem)
+      const functionSelectorSubstring = functionSelector.substring(2)
```
Comment (kmairin) [inline src/filter/filters.ts:195]: also a comment here why it has to be 2. for context, so if someone has to debug the code that could understand

**Review by kmairin — APPROVED** [review]
lgtm just a small nit

File: src/filter/filters.ts (hunk 3 above)
Comment (Quazia) [reply src/filter/filters.ts:195]: Yep can do 😊

Comment (kmairin) [reply src/filter/filters.ts:195]: send itttt

**Pull request merged**